.github/workflows/cd.yaml

```
@@ -0,0 +1,141 @@
+name: Build and publish release artifacts
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5
+        with:
+          go-version-file: go.mod
+          cache: true
+      - name: Run tets
+        run: make test
+      - name: Build helm chart
+        working-directory: charts/eks-pod-identity-agent
+        run: helm package . --version "1.2.3-dev" --app-version "4.5.6-dev"
+
+  build-and-push:
+    name: Build and push Docker container
+    runs-on: ubuntu-latest
+    needs: [test]
+    environment: release
+    env:
+      AWS_REGION_PRIVATE: us-west-2
+      AWS_REGION_PUBLIC: us-east-1
+    permissions:
+      contents: read
+      id-token: write # This is required for requesting the JWT, see https://github.com/aws-actions/configure-aws-credentials#OIDC
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup docker buildx
+        uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
+        
+      - name: Configure AWS credentials for ECR Private
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ env.AWS_REGION_PRIVATE }}
+          role-to-assume: ${{ env.ECR_ROLE }}
+          mask-aws-account-id: 'no'
+
+      - name: Configure AWS credentials for ECR Public
+        if: "startsWith(github.ref, 'refs/tags/v')"
+        uses: aws-actions/configure-aws-credentials@v4
```

```suggestion
        uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4
```

nit: prefer hash over tag